src/Groups/matrices/transform_form.jl

```diff
@@ -0,0 +1,418 @@
+      C,A,e = find_radical(U,F,f,g)
+      # I expect C always to be of rank f
+      C = submatrix(C,1,1,f,f)                     
+      Vprime = star(A)*V*A
+      Z = submatrix(Vprime, 1,1,f,f)
+      Y = submatrix(Vprime,f+1,1,g-f,f)
+      Bprime = submatrix(Vprime,f+1,f+1,g-f,g-f)
+      TR = zero_matrix(F,f,f)
+      D = zero_matrix(F,f,f)
+      for i in 1:f
+         D[i,i] = Z[i,i]
+         for j in i+1:f  TR[i,j] = Z[i,j] end
+      end
+      Aarray = MatElem{T}[]  #TODO type?
+      Barray = MatElem{T}[]
+      for i in 1:f
+         alpha = D[i,i]
+         if alpha != 0
+            push!(Barray, matrix(F,2,2,[-alpha^-1,0,0,alpha]))
+            push!(Aarray, matrix(F,2,2,[1,-alpha^-1,0,1]))
+         else
+            push!(Barray, matrix(F,2,2,[0,1,s,0]))
+            push!(Aarray, matrix(F,2,2,[1,0,0,1]))
+         end
+      end
+      B0,A0 = block_anisotropic_elim(Bprime,_type)
+      B1 = diagonal_join(Barray)
+      B1 = diagonal_join(B1,B0)
+      C = C^-1
+      Temp = vcat(C,-TR*C)
+      Temp = vcat(Temp,-Y*C)
+      Temp = hcat(Temp, vcat(zero_matrix(F,f,g),star(A)))
+      P = zero_matrix(F,2*f,2*f)
+      for i in 1:f
+         P[2*i-1,i] = 1
+         P[2*i,i+f] = 1
+      end
+      A1 = diagonal_join(Aarray)*P
+      A1 = diagonal_join(A1,A0)
+      return B1, A1*Temp
+   else
+      c,f = Int(ceil(d/2)), Int(floor(d/2))
+      B0 = submatrix(B,1,1,c,c)
+      U = submatrix(B,c+1,1,f,c)
+      V = submatrix(B,c+1,c+1,f,f)
+      B1,A0,e = find_radical(B0,F,c,c; e=degF, _is_symmetric=true)
+      B1 = submatrix(B1,1,1,e,e)
+      U = U*star(A0)
+      U1 = submatrix(U,1,1,f,e)
+      U2 = submatrix(U,1,e+1,f,c-e)
+      Z = V-s*U1*B1^-1*star(U1)
+      D1,A1 = block_anisotropic_elim(B1,_type)
+      Temp = zero_matrix(F,d-e,d-e)
+      insert_block!(Temp,s*star(U2),1,c-e+1)
+      insert_block!(Temp,U2,c-e+1,1)
+      insert_block!(Temp,Z,c-e+1,c-e+1)
+      if c-e==0
+         D2,A2 = block_anisotropic_elim(Temp,_type)
+      else
+         D2,A2 = block_anisotropic_elim(Temp, _type; isotr=true, f=c-e)
+      end
+      Temp = hcat(-U1*B1^-1, zero_matrix(F,f,c-e))*A0
+      Temp = vcat(A0,Temp)
+      Temp = insert_block(identity_matrix(F,d),Temp,1,1)
+
+      return diagonal_join(D1,D2), diagonal_join(A1,A2)*Temp
+   end
+end
+
+
+
+# assume B is nondegenerate
+
+
+# returns D, A such that A*B*transpose(frobenius(A)) = D and 
+# D is diagonal matrix (or with blocks [0 1 s 0])
+# f = dimension of the zero block in B in the isotropic case
+function block_herm_elim(B::MatElem{T}, _type) where T <: FieldElem
+   d = nrows(B)
+   F = base_ring(B)
+
+   if d==1
+      return B, identity_matrix(F,1)
+   end
+
+   c = Int(ceil(d/2))
+   B2 = submatrix(B,1,1,c,c)
+   if B2==0
+      D,A = block_anisotropic_elim(B,_type; isotr=true, f=c)
+   else
+      D,A = block_anisotropic_elim(B,_type)
+   end
+
+   return D,A
+end
+
+
+
+# returns D such that D*B*conjugatetranspose(D) is the standard basis
```

Comment:
you call `conjugatetranspose` elsewhere `star` in this file; and in Julia, there is `adjoint`, which is defined as "conjugate transposition" -- though they of course have just complex conjugation in mind, not the unique involution in a finite field of even degree... But we could of course appropriate this for our purposes... just not sure if we should..? @fieker @thofma ?

Reply:
I tested `adjoint` on matrices over finite fields of even degree, but it seems to work just as a transpose.
I used the function `star` because in some cases I need the transpose-conjugate (hermitian) and in other cases I need just the transpose (symmetric and alternating). I find this the best way to avoid code repetitions and unnecessary `if` cycles. I added a line of comment about this in the file.